### PR TITLE
2.9x - Backport some patches

### PR DESCRIPTION
**Pull request by neheb**
The first one fixes the annoying C: issue on Windows

The others are the ones I maintain for OpenWrt.

### CODE REVIEW

**Comment by neheb** [comment]
Build failures on non-linux seem unrelated?

File: web/javascript/common.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ $(document).ready(function() {
 			// Fix min height for isMobileDevice when run in full screen mode from home screen
 			// so the footer appears in the right place
 			$('body div#torrent_container').css('min-height', '338px');
-		$("label[for=torrent_upload_url]").text("URL: ");
+		$("label[for=torrent_upload_url]").text = "URL: ";
```
Comment (glebm) [inline web/javascript/common.js]: This should be reverted, the previous version was correct (`.text()` is a jQuery method)

Comment (neheb) [reply web/javascript/common.js]: Users have reported Safari being fixed by this change.

Comment (glebm) [reply web/javascript/common.js]: This is quite obviously wrong.

Perhaps you're confused by a similar change on master? That one doesn't use jQuery so it doesn't apply here.

**Comment by neheb** [comment]
Nope.

https://github.com/openwrt/packages/issues/4117

**Comment by glebm** [comment]
@neheb The diff you pasted in the comment is completely different from the actual git diff.

This is the git diff: https://github.com/transmission/transmission/commit/bcfe147e6a41ce7774f569ede6179a7ed2c7f560#diff-90cb18eb85291b642ba26b968b3e210e

You comment https://github.com/openwrt/packages/issues/4117#issuecomment-366462120:

> Alternatively for latest stable, this patch should accomplish the same thing:
> (your patch)

I don't think the diff in your comment was ever merged (there is no link there either).

I suggest you try it yourself and read jQuery docs (http://api.jquery.com/text/).

**Comment by neheb** [comment]
Yes it does. That commit cannot be backported as the codebase is quite different.

I own an iOS device and can verify proper behavior.

**Comment by glebm** [comment]
The `$(..).text =` call does absolutely nothing at all though.

So if this fixes some bug, so would removing that line entirely.

**Comment by neheb** [comment]
It's probably some bug in old Safari. I'm on iOS12 where it seems to be fixed. I'll try to verify behavior with other versions later.

File: web/javascript/common.js (hunk 1 above)
Comment (ckerr) [reply web/javascript/common.js]: I'm OK with all the other backport changes, but this doesn't actually seem to be a backport and I agree with @glebm that this change seems to be in error. [text(text)](http://api.jquery.com/text/#text2) is indeed a jQuery method that's used in [other parts of the 2.9x web client](https://github.com/transmission/transmission/blob/2.9x/web/javascript/transmission.js#L1266).

I'm :+1: on this PR if this one-liner is removed.

Comment (neheb) [reply web/javascript/common.js]: Removed. I think whatever bug this fixes has long since been fixed on Apple's side.

**Comment by neheb** [comment]
Removed the offending commit and rebased.

**Comment by neheb** [comment]
Meh.

**Pull request closed without merging**